python/cugraph/proto/components/scc.py

```diff
@@ -0,0 +1,165 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import cudf
+import cugraph
+import numpy as np
+
+
+#
+# TRIM Process:
+#   - removed single vertex componenets
+#   - select vertex with highest out degree
+#   - forwards BFS
+#   - backward BFS
+#   - compute intersection = components
+#   - remove component
+#   - repeat
+
+
+def strong_connected_component(source, destination):
+    """
+    Generate the strongly connected components
+    using the FW-BW-TRIM approach, but skipping the trimming)
+
+    Parameters
+    ----------
+    source : cudf.Seriers
+        A cudf seriers that contains the source side of an edge list
+
+    destination : cudf.Seriers
+        A cudf seriers that contains the destination side of an edge list
+
+    Returns
+    -------
+    cdf : cudf.DataFrame - a dataframe for components
+        df['vertex']   - the vertex ID
+        df['id']       - the component ID
+
+    sdf : cudf.DataFrame - a dataframe with single vertex components
+        df['vertex']   - the vertex ID
+
+    count - int - the number of components found
+
+
+    Examples
+    --------
+    >>> M = read_mtx_file(graph_file)
```

Comment:
@afender requested to replace read_mtx_file in the examples with csv read.

Here's an example.
```
    >>> M = cudf.read_csv('datasets/karate.csv', delimiter=' ',
    >>>                   dtype=['int32', 'int32', 'float32'], header=None)
    >>> sources = cudf.Series(M['0'])
    >>> destinations = cudf.Series(M['1'])
    >>> G = cugraph.Graph()
    >>> G.add_edge_list(sources, destinations, None)
    >>> source_col, dest_col, numbering_map = cugraph.renumber(sources,
    >>>                                                        destinations)
```